core/network/portrange.go

```diff
@@ -95,7 +95,7 @@ func (grp GroupedPortRanges) UniquePortRanges() []PortRange {
 
 // Clone returns a copy of this port range grouping.
 func (grp GroupedPortRanges) Clone() GroupedPortRanges {
-	if len(grp) == 0 {
+	if grp == nil {
 		return nil
 	}
```

Comment:
```suggestion
	if len(grp) == 0 {
		return make(GroupedPortRanges)
	}
```

Reply:
Disagree.

Because then if you pass in nil, you will get an empty map

Reply:
Then don't pass nil.

Reply:
I can't control what other people pass into my function!